src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -264,7 +260,16 @@ func (r *ClusterReconciler) Reconcile(
 		log.Info(requeueErr.Error())
 		return ctrl.Result{RequeueAfter: requeueErr.RequeueAfter}, nil
 	}
-	return ctrl.Result{}, err
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// setting license should be at the last part as it requires AdminAPI to be running
+	if err := r.setLicense(ctx, &redpandaCluster, log); err != nil {
+		log.Info(fmt.Sprintf("setting license: %s", err.Error()))
```

Comment:
why is this not error? also can you say something like "error setting license, will retry in 1 minute.."?

Reply:
because it's expected to happen at some point (i.e. on first reconciliation while cluster is not yet ready) but I've updated it to check for that :pray: